### PR TITLE
Reimbursements: Improved feature flags

### PR DESCRIPTION
**Pull request by YodaLightsabr**
## Summary of the problem
Feature flags are user level

## Describe your changes
Feature flags are now org level and can only be enabled by users with the feature flag enabled

Also added a card to the org settings page only visible to users with the feature flag enabled

### CODE REVIEW

File: app/views/events/edit.html.erb [hunk 1]
```diff
@@ -61,6 +61,40 @@
...
+  <%# Only allow users with access enable this on orgnizations and allow orgs with it enabled to turn it off %>
+  <div class="card pb0 mb3" id="reimbursements">
+    <div class="card__banner card__banner--top flex justify-between items-center feature-reimbursements <%= "b--info" if Flipper.enabled?(:reimbursements_2024_03_10, @event) %> ">
+      <h3 class="h1 pt2 pb2 my0 color-black">Reimbursements 2.0</h3>
+      <% if Flipper.enabled?(:reimbursements_2024_03_10) %>
+        <%# Feature is enabled for all users, don't include a choice %>
+        <div class="actions tooltipped tooltipped--w inline-block mt1" aria-label="This feature is now generally available!">
+          <span class="btn bg-accent disabled">Shipped!</span>
+        </div>
+        <%# link_to "Shipped!", enable_feature_path(feature: :transfers_2022_04_21), method: :post, class: 'btn bg-accent' %>
```
Comment (sarthaktexas) [inline app/views/events/edit.html.erb]: Why is this a comment?

File: app/views/events/edit.html.erb [hunk 2]
```diff
@@ -61,6 +61,40 @@
...
+      <% elsif Flipper.enabled?(:reimbursements_2024_03_10, @event) %>
+        <%# Feature is enabled, let the user opt-out %>
+        <%= link_to "Disable", disable_feature_event_path(feature: :reimbursements_2024_03_10), method: :post, class: "btn bg-accent #{"disabled" unless policy(@event).disable_feature?} tooltipped tooltipped--w", "aria-label": "Only managers can disable feature previews." %>
+      <% else %>
+        <%# Feature is disabled, let the user opt-in %>
+        <%= link_to "Enable", enable_feature_event_path(feature: :reimbursements_2024_03_10), method: :post, class: "btn bg-info #{"disabled" unless policy(@event).enable_feature?} tooltipped tooltipped--w", "aria-label": "Only managers can enable feature previews." %>
+      <% end %>
+    </div>
+    <p>
+      Reimburse team members and let volunteer submit epenses directly in HCB!
```
Comment (sampoder) [inline app/views/events/edit.html.erb]:
```suggestion
      Reimburse team members and let volunteer submit expenses directly in HCB!
```

Comment (sarthaktexas) [inline app/views/events/edit.html.erb]:
```suggestion
      Reimburse team members and let volunteer submit expenses directly in HCB!
```

File: app/views/events/edit.html.erb [hunk 3]
```diff
@@ -61,6 +61,40 @@
...
+      <% elsif Flipper.enabled?(:reimbursements_2024_03_10, @event) %>
+        <%# Feature is enabled, let the user opt-out %>
+        <%= link_to "Disable", disable_feature_event_path(feature: :reimbursements_2024_03_10), method: :post, class: "btn bg-accent #{"disabled" unless policy(@event).disable_feature?} tooltipped tooltipped--w", "aria-label": "Only managers can disable feature previews." %>
+      <% else %>
+        <%# Feature is disabled, let the user opt-in %>
+        <%= link_to "Enable", enable_feature_event_path(feature: :reimbursements_2024_03_10), method: :post, class: "btn bg-info #{"disabled" unless policy(@event).enable_feature?} tooltipped tooltipped--w", "aria-label": "Only managers can enable feature previews." %>
+      <% end %>
+    </div>
+    <p>
+      Reimburse team members and let volunteer submit expenses directly in HCB!
```
Comment (sampoder) [inline app/views/events/edit.html.erb]:
```suggestion
      Reimburse team members and volunteers directly through HCB! It's our very own version of Expensify.
```

Thoughts on this copy?

**Review by sarthaktexas — CHANGES_REQUESTED** [review]

File: app/views/users/_nav.html.erb [hunk 4]
```diff
@@ -26,7 +26,7 @@
       tooltip: "See expense reimbursements",
       icon: "attachment",
       async_badge: my_draft_reimbursements_icon_path,
-      selected: local_assigns[:selected] == :reimbursements if Flipper.enabled?(:reimbursements_2024_03_10, current_user) %>
+      selected: local_assigns[:selected] == :reimbursements if current_user.events.flag_enabled(:reimbursements_2024_03_10).exists? %>
```
Comment (YodaLightsabr) [inline app/views/users/_nav.html.erb]:
```suggestion
      selected: local_assigns[:selected] == :reimbursements if current_user.events.flag_enabled(:reimbursements_2024_03_10).exists? || Flipper.enabled?(:reimbursements_2024_03_10) %>
```

Check for any of users orgs having it _or_ for general availability

**Review by sarthaktexas — DISMISSED** [review]

**Review by sampoder — DISMISSED** [review]

**Review by sampoder — APPROVED** [review]
Code LGTM!

**Pull request merged**